app-creator/src/widgets/dropzone-widget/dropzone-widget.ts

```diff
@@ -168,15 +166,16 @@ export class Dropzone extends LitElement {
    * @param widget widget currently being dragged.
    * @param y the y coordinate of the triggered event.
    */
-  getNextElement(widget: HTMLElement, y: number): Element | null {
+  getNextElement(widget: Element, y: number): Element | null {
     const container = this.shadowRoot?.getElementById(CONTAINER_ID);
     if (container == null) {
       return null;
     }
 
     const children = Array.from(container.children).filter(
       (child) =>
-        child.id !== `${widget.id}-${store.widgetIDs[widget.id]}-wrapper`
+        child.id !==
+        `${widget.id}-${store.getState().widgetIDs[widget.id]}-wrapper`
```

Comment:
Could you add a comment about what we are filtering? Could we achieve the same result with !child.id.includes('wrapper')?

Reply:
So here we are grabbing all the child widgets that are not the currently dragged widget. Since the direct children of the dropzone are the draggable wrappers. We are checking for the wrapper of the currently dragged widget. Therefore we need to check  `${widget.id}-${store.getState().widgetIDs[widget.id]}-wrapper`.

Reply:
I just pushed an update. I'm still trying out a refactor based on @sufyanAbbasi comment:

> Same could be true about dragging and editing widget, both of these fields should be mutually exclusive, since we don't expect a user to be dragging and editing a widget at the same time.
> 
> For example, the store can simply be:
> 
> ```
> {
>   element: Element | null; // The main widget being interacted with.
>   eventType: "dragging"|"editing"|"reordering"|"adding"
>   widgetIDs: ...
> }
> ```
https://github.com/googleinterns/earthengine-community/pull/5#discussion_r433396091